.circleci/config.yml

```
@@ -25,8 +25,7 @@ commands:
             curl -LsSf https://astral.sh/uv/install.sh | sh
             uv venv
             source .venv/bin/activate
-            uv pip install .
-            uv pip install -r ./test_requirements/requirements_core.txt
+            uv sync --extra dev_core
```

Recommend adding a `uv lock --check` step here (or anywhere else in the CI that makes sense, maybe even in its own job?) -- to ensure that the lockfile matches what's listed in the `pyproject.toml`.

```suggestion
            uv lock --check
            uv sync --extra dev_core
```